mqtt/mqtt-edgehub/src/command.rs

```diff
@@ -1,200 +1,116 @@
-) -> Result<(), CommandHandlerError> {
-    debug!("command handler subscribing to disconnect topic");
-    let subscriptions = topics.iter().map(|topic| proto::SubscribeTo {
-        topic_filter: topic.to_string(),
-        qos: proto::QoS::AtLeastOnce,
-    });
-
-    for subscription in subscriptions {
-        client
-            .subscribe(subscription)
-            .map_err(CommandHandlerError::SubscribeFailure)?;
+impl Command for AuthorizedIdentities {
+    fn topic(&self) -> String {
+        self.topic.clone()
     }
 
-    let mut subacks: HashSet<_> = topics.iter().map(Clone::clone).collect();
-
-    while let Some(event) = client
-        .try_next()
-        .await
-        .map_err(CommandHandlerError::PollClientFailure)?
-    {
-        if let Event::SubscriptionUpdates(subscriptions) = event {
-            for subscription in subscriptions {
-                if let SubscriptionUpdateEvent::Subscribe(sub) = subscription {
-                    subacks.remove(&sub.topic_filter);
-                }
-            }
-
-            if subacks.is_empty() {
-                debug!("command handler successfully subscribed to disconnect topic");
-                return Ok(());
-            }
+    fn handle(
+        &self,
+        broker_handle: &mut BrokerHandle,
+        publication: &ReceivedPublication,
+    ) -> Result<(), HandleEventError> {
+        let array: ServiceIdentity = serde_json::from_slice(&publication.payload)
+            .map_err(HandleEventError::ParseClientId)?;
+        if let Err(e) = broker_handle.send(Message::System(SystemEvent::AuthorizationUpdate(
+            Box::new(array),
+        ))) {
+            return Err(HandleEventError::SendAuthorizedIdentitiesToBroker(e));
         }
-    }
 
-    error!("command handler failed to subscribe to disconnect topic");
-    Err(CommandHandlerError::MissingSubacks(
-        subacks.into_iter().collect::<Vec<_>>(),
-    ))
+        info!("succeeded sending authorized identity scopes to broker",);
+        Ok(())
+    }
 }
 
-#[derive(Debug, thiserror::Error)]
-pub enum CommandHandlerError {
-    #[error("failed to receive expected subacks for command topics: {0:?}")]
-    MissingSubacks(Vec<String>),
-
-    #[error("failed to subscribe command handler to command topic")]
-    SubscribeFailure(#[from] UpdateSubscriptionError),
-
-    #[error("failed to poll client when validating command handler subscriptions")]
-    PollClientFailure(#[from] mqtt3::Error),
+pub fn init_commands() -> HashMap<String, Box<dyn Command + Send>> {
+    let mut commands: HashMap<String, Box<dyn Command + Send>> = HashMap::new();
+    commands.insert(
+        DISCONNECT_TOPIC.to_string(),
+        Box::new(Disconnect {
+            topic: DISCONNECT_TOPIC.to_string(),
+        }),
+    );
+    commands.insert(
+        AUTHORIZED_IDENTITIES_TOPIC.to_string(),
+        Box::new(AuthorizedIdentities {
+            topic: AUTHORIZED_IDENTITIES_TOPIC.to_string(),
+        }),
```

Comment:
This needs to be in a Box because commands is a hashmap where values are implementors of trait Command, which does not have a determined size?

Reply:
That is correct. Talked with Denis offline. It's ok for now since these Commands won't be a hot path. If they need to be in the future, we can change it then